Capstone Project/WeightLossViewController.m

```diff
@@ -73,5 +95,70 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 
 }
 
+- (void) getPosts{
+    PFQuery *postQuery = [Post query];
+    [postQuery whereKey:@"author" equalTo:[PFUser currentUser]];
+    
+    [postQuery findObjectsInBackgroundWithBlock:^(NSArray<Post *> * _Nullable posts, NSError * _Nullable error) {
+        if (posts) {
+            // do something with the data fetched
+            self.arrayOfPosts = (NSMutableArray *)posts;
+
+            [self.collectionView reloadData];
+
+        }
+        else {
+            // handle error
+            NSLog(@"%@", error.localizedDescription);
+            UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Cannot Get Posts"
+                                           message:@"The internet connection appears to be offline."
+                                           preferredStyle:UIAlertControllerStyleAlert];
+             
+            UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+               handler:^(UIAlertAction * action) {}];
+             
+            [alert addAction:defaultAction];
+            [self presentViewController:alert animated:YES completion:nil];
+        }
+        [self.refreshControl endRefreshing];
+
+    }];
+    
+}
+
+- (void) didPost{
+    
+    [self getPosts];
+    [self.collectionView reloadData];
```

Comment:
There is no need for reloading it here again. Line 107 loads it. Reloading it here has no use